lib/puma/cluster.rb

```diff
@@ -91,11 +92,8 @@ def spawn_workers
       end
 
       if @options[:fork_worker] && all_workers_in_phase?
-        @fork_writer << "0\n"
-
-        if worker_at(0).phase > 0
```

Comment:
I am not sure it makes sense to run these hooks after the server has been restarted; if there is anything in there meant to reset server state it would need to go first, so I've bumped it in front of the restart command.

Reply:
indeed, "after the server has temporarily stopped serving requests" is the specific timing I based this off of, which would be a very weird way to describe "and also after it's started serving them again"